app/controllers/BalanceRequestController.scala

```diff
@@ -47,40 +54,88 @@ class BalanceRequestController @Inject() (
   val runtime: IORuntime
 ) extends BackendController(cc)
     with IOActions
+    with Logging
+    with ErrorLogging
     with LocationWithContext {
 
+  private val AcceptHeaderRegex = """application/vnd\.hmrc\.(1.0)\+json""".r
+
+  private def requireAcceptHeader[A](
+    result: => IO[Result]
+  )(implicit request: Request[A]): IO[Result] =
+    request.headers
+      .get(HeaderNames.ACCEPT)
+      .flatMap(AcceptHeaderRegex.findFirstIn)
+      .map(_ => result)
+      .getOrElse {
+        val error     = MissingAcceptHeaderError()
+        val errorJson = Json.toJson(error)
+        IO.pure(NotAcceptable(errorJson))
+      }
+
   def submitBalanceRequest: Action[BalanceRequest] =
     authenticate().io(parse.json[BalanceRequest]) { implicit request =>
-      service.submitBalanceRequest(request.body).map {
-        case Right(Right(success @ BalanceRequestSuccess(_, _))) =>
-          Ok(Json.toJson(PostBalanceRequestSuccessResponse(success)))
+      requireAcceptHeader {
+        service
+          .submitBalanceRequest(request.body)
+          .flatTap(logServiceError("submitting balance request", _))
+          .map {
+            case Right(Right(success @ BalanceRequestSuccess(_, _))) =>
+              Ok(Json.toJson(PostBalanceRequestSuccessResponse(success)))
 
-        case Right(Right(error @ BalanceRequestFunctionalError(_))) =>
-          BadRequest(Json.toJson(PostBalanceRequestFunctionalErrorResponse(error)))
+            case Right(Right(error @ BalanceRequestFunctionalError(_))) =>
+              BadRequest(Json.toJson(PostBalanceRequestFunctionalErrorResponse(error)))
 
-        case Right(Left(_)) =>
-          // Accepted(Json.toJson(PostBalanceRequestPendingResponse(balanceId))).withHeaders(
-          //   HeaderNames.LOCATION -> routes.BalanceRequestController
-          //     .getBalanceRequest(balanceId)
-          //     .pathWithContext
-          // )
-          val error     = UpstreamTimeoutError()
-          val errorJson = Json.toJson[BalanceRequestError](error)
-          GatewayTimeout(errorJson)
+            case Right(Left(_)) =>
+              // Accepted(Json.toJson(PostBalanceRequestPendingResponse(balanceId))).withHeaders(
+              //   HeaderNames.LOCATION -> routes.BalanceRequestController
+              //     .getBalanceRequest(balanceId)
+              //     .pathWithContext
+              // )
```

Comment:
Is this meant to be here?

Reply:
Yes, we are currently not sure how this will be implemented because documenting multiple response codes is not supported by the Developer Hub API doc rendering. 
Unfortunately this might mean that we actually don't support returning an asynchronous response when the user's request times out and return a gateway timeout instead.
The backend code still supports this option for the users so I have only commented out the code for now.

Reply:
https://jira.tools.tax.service.gov.uk/browse/APSR-1358 for more info